src/main/kotlin/plus_calculator/domain/StringAddCalculator.kt

```diff
@@ -0,0 +1,47 @@
+package plus_calculator.domain
+
+class StringAddCalculator {
+    companion object {
+        private const val ZERO = 0
+        private val CUSTOM_DELIMITER = Regex("//(.)\n(.*)")
+        private val DEFAULT_DELIMITER = Regex("[,:]")
+    }
+    fun add(text: String?): Int {
+        if (text.isNullOrBlank()) {
+            return ZERO
+        }
+
+        val textOperands: List<String> = this.tokenizesText(text)
+
+        return textOperands.sumOf { parseNumber(it) }
+    }
+
+    private fun tokenizesText(text: String): List<String> {
+        val matcher = CUSTOM_DELIMITER.find(text)
+
+        if (matcher != null) {
+            val (customDelimiter: String, matchedText: String) = matcher.destructured
+
+            return matchedText.split(customDelimiter)
+        }
+
+        return text.split(DEFAULT_DELIMITER)
+    }
+
+    private fun parseNumber(numberString: String): Int {
+        try {
+            val number = numberString.toInt()
```

Comment:
```suggestion
        val number = numberString.toIntOrNull() ?: throw RuntimeException("숫자만 가능합니다.")
```
코틀린의 많은 확장함수들은 `xxxOrNull`과 같이 예외인 경우 null을 반환하는 꼴을 지원합니다. 
이런 함수들을 사용할 경우 `try~catch` 또는 `runCatching` 등의 방법을 사용하지 않고도 예외처리를 할 수 있어요 😄